frontend/src/metabase/public/containers/PublicOrEmbeddedDashboard/PublicOrEmbeddedDashboard.tsx

```diff
@@ -98,16 +98,18 @@ class PublicOrEmbeddedDashboardInner extends Component<PublicOrEmbeddedDashboard
       dashboardId,
     } = this.props;
 
-    initialize();
+    if (!this.props.dashboard || isForceUpdate) {
```

Comment:
I'd rather we use the current approach for the same function on our dashboard app.

For example
https://github.com/metabase/metabase/blob/ee7c434781cf61bce03edc5494700e75283fc092/frontend/src/metabase/dashboard/components/Dashboard/Dashboard.tsx#L263

This could be achieve here with the same flag, but without wrapping this whole thing inside another if block
```ts
    const clearCache = Boolean(!dashboard || isForceUpdate);

    initialize({ clearCache });

    const result = await fetchDashboard({
      dashId: String(dashboardId),
      queryParams: parameterQueryParams,
      options: {
        clearCache,
      },
    });
    ...
      if (this.props.dashboard?.tabs?.length === 0) {
        await fetchDashboardCardData({ reload: false, clearCache });
      }
```

The goal is to not try to deviate from the counterpart entity we have in the main app, so it's easier to compare feature parity and maintain them.

Reply:
Apart from the comment above this code still has a problem that it won't load a new dashboard if you unmount and remount the dashboard again. Since `_initialize()` will be called without `true`
https://github.com/metabase/metabase/blob/ee7c434781cf61bce03edc5494700e75283fc092/frontend/src/metabase/public/containers/PublicOrEmbeddedDashboard/PublicOrEmbeddedDashboard.tsx#L125-L127

I think the logic should be
- By default reload the dashboard
- Only _not_ reloading the dashboard _if_ navigating via back button

We're doing the second one in the main app with this flag
https://github.com/metabase/metabase/blob/ee7c434781cf61bce03edc5494700e75283fc092/frontend/src/metabase/dashboard/components/Dashboard/Dashboard.tsx#L263

However, we need to achieve the same result here somehow. But using `dashboard` isn't working because we don't know if a render is from navigating back via the back button, or if this is the first render e.g. from choosing the dashboard in the analytics overview tab
![image](https://github.com/metabase/metabase/assets/1937582/c3d1c179-c9ee-40a2-a510-c372824fa13b)


In short, these 2 comments are 2 separate issues.
1. I want consistent code
2. The code for caching dashboard when navigating back to the same dashboard doesn't quite work.